bridged-tokens-wrapper/src/dfp_big_uint.rs

```diff
@@ -0,0 +1,43 @@
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+#[derive(Clone, PartialEq, Eq)]
+pub struct DFPBigUint<M: ManagedTypeApi> {
+    bu: BigUint<M>,
+    num_decimals: u32,
+}
+
+impl<M: ManagedTypeApi> DFPBigUint<M> {
+    pub fn from_raw(bu: BigUint<M>, num_decimals: u32) -> Self {
+        DFPBigUint { bu, num_decimals }
+    }
+
+    pub fn convert(&self, decimals: u32) -> Self {
+        if self.num_decimals < decimals {
+            let diff_decimals = decimals - self.num_decimals;
+            return DFPBigUint {
+                bu: self.bu.clone() * 10u64.pow(diff_decimals),
+                num_decimals: decimals,
+            };
+        }
+        if self.num_decimals > decimals {
+            let diff_decimals = self.num_decimals - decimals;
+            return DFPBigUint {
+                bu: self.bu.clone() / 10u64.pow(diff_decimals),
+                num_decimals: decimals,
+            };
+        }
+        self.clone()
+    }
```

Comment:
This function can be made more efficient:
- consume `self` and let the caller clone if they need to
- use early exit conditions (it's the most common, and it also makes the code more readable)

```suggestion
    pub fn convert(self, decimals: u32) -> Self {
            if self.num_decimals == decimals {
                 return self;
            }
    
        let new_bu=  if self.num_decimals > decimals { 
             let diff_decimals = self.num_decimals - decimals;
             self.bu / 10u64.pow(diff_decimals)
         else { 
               let diff_decimals = decimals - self.num_decimals;
               self.bu * 10u64.pow(diff_decimals)
         };
         
         DFPBigUint {
                bu: new_bu,
                num_decimals: decimals,
            }
```